File: src/aws/lambdas/incremental_distribution/src/test/java/uk/nhs/nhsx/core/auth/StandardAuthenticationTest.java
package uk.nhs.nhsx.core.auth;

import org.junit.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class StandardAuthenticationTest {


    @Test
    public void anApi() throws Exception {
        assertSomeKeyNames(api());
    }

    @Test
    public void mobileApi() throws Exception {
        assertSomeKeyNames(api());
    }

    private void assertSomeKeyNames(ApiKeyAuthorizer authorizer) {
        assertThat(authorizer.authorize(ApiKey.of("mobile", "")), is(true));
        assertThat(authorizer.authorize(ApiKey.of("blah_c-_134", "")), is(true));
        assertThat(authorizer.authorize(ApiKey.of("third-party-integration-2020102", "")), is(true));
        assertThat(authorizer.authorize(ApiKey.of("-/_+=.@!", "")), is(false));
        assertThat(authorizer.authorize(ApiKey.of("©∞", "")), is(false));
        assertThat(authorizer.authorize(ApiKey.of("0123456789012345678900123456789012345678901234567890", "")), is(false));
    }

    private ApiKeyAuthorizer api() {
        return StandardAuthentication.apiKeyNameValidator();
    }
}

File: src/aws/lambdas/incremental_distribution/src/test/java/uk/nhs/nhsx/core/auth/StandardAuthenticationTest.kt
package uk.nhs.nhsx.core.auth

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test

class StandardAuthenticationTest {

    @Test
    fun anApi() {
        assertSomeKeyNames(api())
    }

    @Test
    fun mobileApi() {
        assertSomeKeyNames(api())
    }

    private fun assertSomeKeyNames(authorizer: ApiKeyAuthorizer) {
        assertThat(authorizer.authorize(ApiKey("mobile", "")), `is`(true))
        assertThat(authorizer.authorize(ApiKey("blah_c-_134", "")), `is`(true))
        assertThat(authorizer.authorize(ApiKey("third-party-integration-2020102", "")), `is`(true))
        assertThat(authorizer.authorize(ApiKey("-/_+=.@!", "")), `is`(false))
        assertThat(authorizer.authorize(ApiKey("©∞", "")), `is`(false))
        assertThat(authorizer.authorize(ApiKey("0123456789012345678900123456789012345678901234567890", "")), `is`(false))
    }

    private fun api(): ApiKeyAuthorizer = StandardAuthentication.apiKeyNameValidator()
}
